File: Forms/src/0_contracts/format-schema.ts
import { z } from 'zod';

export const FormatFieldSchema = z.object({
  key: z.string(),
  label: z.string(),
  prefix: z.string().default(''),
  suffix: z.string().default(''),
  fallbackTo: z.string().optional(),
  required: z.boolean().default(false),
});

export type FormatField = z.infer<typeof FormatFieldSchema>;

export const FormatSchemaDefinition = z.object({
  id: z.string(),
  name: z.string(),
  icon: z.string().default('📋'),
  description: z.string().optional(),
  headerTemplate: z.string().default(''),
  footerTemplate: z.string().default(''),
  defaultValues: z.record(z.string(), z.string()).default({}),
  fields: z.array(FormatFieldSchema),
});

export type FormatSchema = z.infer<typeof FormatSchemaDefinition>;

File: app_native_desktop/app_forms/1_Backend/Contracts/Entities/FormatSchema.cs
namespace AppForms.Backend.Contracts.Entities;

public record FormatField
{
    public string Key { get; init; } = string.Empty;
    public string Label { get; init; } = string.Empty;
    public string Prefix { get; init; } = string.Empty;
    public string Suffix { get; init; } = string.Empty;
    public string? FallbackTo { get; init; }
    public bool Required { get; init; } = false;
}

public record FormatSchema
{
    public string Id { get; init; } = string.Empty;
    public string Name { get; init; } = string.Empty;
    public string Icon { get; init; } = "📋";
    public string Description { get; init; } = string.Empty;
    public string HeaderTemplate { get; init; } = string.Empty;
    public string FooterTemplate { get; init; } = string.Empty;
    public Dictionary<string, string> DefaultValues { get; init; } = new(StringComparer.OrdinalIgnoreCase);
    public List<FormatField> Fields { get; init; } = new();
}

public record ConversionItem
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public string RawInput { get; init; } = string.Empty;
    public LeadEntity Lead { get; init; } = new();
    public Dictionary<string, string> FormattedOutputs { get; init; } = new();
    public string SelectedSchemaId { get; init; } = string.Empty;
    public DateTime ConvertedAt { get; init; } = DateTime.Now;
}
